src/scanners.rs

```diff
@@ -456,6 +456,143 @@ pub fn calc_indent(text: &str, max: usize) -> (usize, usize) {
     (i, spaces)
 }
 
+pub struct RefDefScan<'a> {
+    pub(crate) bytecount: usize,
+    pub(crate) label: &'a str,
+    pub(crate) dest: &'a str,
+    pub(crate) title: Option<&'a str>
+}
+
+pub fn scan_refdef(data: &str) -> Option<RefDefScan> {
+    let mut i = scan_whitespace_no_nl(data);
+
+    // defs can only be indented up to three spaces
```

Comment:
It seems unlikely this will handle tabs correctly. I'm 90% certain you want to use the `LineStart` mechanism for initial indent, and assume all initial whitespace has been consumed by the time you get to the ref scanning.